src/Native/Runtime/windows/CoffNativeCodeManager.cpp

```diff
@@ -380,9 +380,70 @@ void CoffNativeCodeManager::EnumGcRefs(MethodInfo *    pMethodInfo,
 
 UIntNative CoffNativeCodeManager::GetConservativeUpperBoundForOutgoingArgs(MethodInfo * pMethodInfo, REGDISPLAY * pRegisterSet)
 {
-    // @TODO: CORERT: GetConservativeUpperBoundForOutgoingArgs
+#if defined(_TARGET_AMD64_)
+
+    // Return value
+    UIntNative upperBound;
+    CoffNativeMethodInfo* pNativeMethodInfo = (CoffNativeMethodInfo *) pMethodInfo;
+    
+    // Get the method's GC info
+    size_t unwindDataBlobSize;
+    PTR_VOID pUnwindDataBlob = GetUnwindDataBlob(m_moduleBase, pNativeMethodInfo->runtimeFunction, &unwindDataBlobSize);
+    PTR_UInt8 p = dac_cast<PTR_UInt8>(pUnwindDataBlob) + unwindDataBlobSize;
+    uint8_t unwindBlockFlags = *p++;
+    GcInfoDecoder decoder(GCInfoToken(p), DECODE_REVERSE_PINVOKE_VAR);
+    
+    UINT32 stackBasedRegister = decoder.GetStackBaseRegister();
+    
+    // Initialize the base pointer to either  stack or frame pointer
+    if(stackBasedRegister == NO_STACK_BASE_REGISTER) 
+    {
+        // Unwind the current method context to the caller's context to get its stack pointer
+        // and obtain the upper bound of the calee is the value just below the caller's return address on the stack
+        SIZE_T  EstablisherFrame;
+        PVOID   HandlerData;
+        CONTEXT context;
+        context.Rsp = pRegisterSet->GetSP();
+        context.Rbp = pRegisterSet->GetFP();
+        context.Rip = pRegisterSet->GetIP();
+
+        RtlVirtualUnwind(NULL,
+                        dac_cast<TADDR>(m_moduleBase),
+                        pRegisterSet->IP,
+                        (PRUNTIME_FUNCTION)pNativeMethodInfo->runtimeFunction,
+                        &context,
+                        &HandlerData,
+                        &EstablisherFrame,
+                        NULL);
+
+        upperBound = dac_cast<TADDR>(context.Rsp - 2 * sizeof (PVOID));
```

Comment:
What is `2 *` for? I think one slot is for the pushed return address. But what is the other one for?